android/src/main/java/com/thebluealliance/androidclient/viewmodels/ViewModelRenderer.java

```diff
@@ -0,0 +1,11 @@
+package com.thebluealliance.androidclient.viewmodels;
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+import android.support.annotation.WorkerThread;
+
+public interface ViewModelRenderer<VIEWMODEL, ARGS> {
+
+    @WorkerThread
+    @Nullable VIEWMODEL renderToViewModel(Context context, @Nullable ARGS args);
```

Comment:
There's virtually no way to do this kind of thing generically. A ViewModel is really just a nice term for some arbitrary object that will be passed to a view and bound to it later. There's pretty tight coupling between a ViewModel and the view that will ultimately display it later.